python/lsst/daf/butler/cli/butler.py

```diff
@@ -278,20 +300,24 @@ def _mergeCommandLists(
         return a
 
     @classmethod
-    def _getPluginCommands(cls) -> defaultdict[str, list[str]]:
+    def _getPluginCommands(cls) -> defaultdict[str, list[str | PluginCommand]]:
         """Get the commands offered by plugin packages.
 
         Returns
         -------
         commands : `defaultdict` [`str`, `list` [`str`]]
             The key is the command name. The value is a list of package(s) that
             contains the command.
+
+        Notes
+        -----
+        Assumes that if entry points are defined, the plugin environment
+        variable will not be defined for that same package.
```

Comment:
Not sure if I read it right, but bot `pie_base` and `daf_butler_migrate` add entry point while keeping envvar too?

Reply:
Correct. Because scons does not create the metadata file so python can't find the entry points. There is another ticket to see if scons can be modified to write the correct files so that this all works without involving EUPS. For now we assume that either the package is pip installed or EUPS is setting the env var.